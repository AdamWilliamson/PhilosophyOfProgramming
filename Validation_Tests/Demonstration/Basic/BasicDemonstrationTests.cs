using ApprovalTests;
using System.Threading.Tasks;
using Validations_Tests.TestHelpers;
using Xunit;

namespace Validations_Tests.Demonstration.Basic;

public class BasicDemonstrationTests
{
    [Fact]
    public async Task BasicValidator_Validate_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicSongValidator());
        var song = new BasicSong(
            "Disturbed", 
            null, 
            "Down With The Sickness", 
            2.3, 
            string.Empty
        );

        // Act
        var results = await runner.Validate(song);
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }

    [Fact]
    public void BasicValidator_Describe_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicSongValidator());

        // Act
        var results = runner.Describe();
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }
}

