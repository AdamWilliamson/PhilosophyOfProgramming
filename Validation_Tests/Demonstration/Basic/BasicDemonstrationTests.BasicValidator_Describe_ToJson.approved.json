{
    "results": [
        {
            "property": "TrackNumber",
            "outcomes": [
                {
                    "propertyName": "TrackNumber",
                    "message": "Must not be null",
                    "group": null
                }
            ]
        },
        {
            "property": "TrackName",
            "outcomes": [
                {
                    "propertyName": "TrackName",
                    "message": "Must equal to 'Definitely Not The Correct Song Name.'",
                    "group": null
                }
            ]
        },
        {
            "property": "Duration",
            "outcomes": [
                {
                    "propertyName": "Duration",
                    "message": "Songs must force you to travel slowly backwards in time to listen to.",
                    "group": null
                }
            ]
        }
    ]
}